src/auto/pixbuf.rs

```diff
@@ -0,0 +1,531 @@
+    //    unsafe { TODO: call ffi::gdk_pixbuf_new_from_stream() }
+    //}
+
+    //pub fn new_from_stream_at_scale<'a, P: IsA</*Ignored*/gio::InputStream>, Q: Into<Option<&'a gio::Cancellable>>>(stream: &P, width: i32, height: i32, preserve_aspect_ratio: bool, cancellable: Q) -> Result<Pixbuf, Error> {
+    //    unsafe { TODO: call ffi::gdk_pixbuf_new_from_stream_at_scale() }
+    //}
+
+    pub fn new_from_xpm_data(data: &[&str]) -> Pixbuf {
+        unsafe {
+            from_glib_full(ffi::gdk_pixbuf_new_from_xpm_data(data.to_glib_none().0))
+        }
+    }
+
+    //pub fn from_pixdata(pixdata: /*Ignored*/&Pixdata, copy_pixels: bool) -> Result<Pixbuf, Error> {
+    //    unsafe { TODO: call ffi::gdk_pixbuf_from_pixdata() }
+    //}
+
+    pub fn get_file_info(filename: &str) -> (Option<PixbufFormat>, i32, i32) {
+        unsafe {
+            let mut width = mem::uninitialized();
+            let mut height = mem::uninitialized();
+            let ret = from_glib_none(ffi::gdk_pixbuf_get_file_info(filename.to_glib_none().0, &mut width, &mut height));
+            (ret, width, height)
+        }
+    }
+
+    #[cfg(any(feature = "v2_32", feature = "dox"))]
+    pub fn get_file_info_async<'a, P: Into<Option<&'a gio::Cancellable>>, Q: Fn(Result<(i32, i32), Error>) + Send + Sync + 'static>(filename: &str, cancellable: P, callback: Q) {
+        let cancellable = cancellable.into();
+        let cancellable = cancellable.to_glib_none();
+        let user_data: Box<Box<Fn(Result<(i32, i32), Error>) + Send + Sync + 'static>> = Box::new(Box::new(callback));
+        extern "C" fn get_file_info_async_trampoline(_source_object: *mut gobject_ffi::GObject, res: *mut gio_ffi::GAsyncResult, user_data: glib_ffi::gpointer)
+        {
+            callback_guard!();
+            unsafe {
+                let mut error = ptr::null_mut();
+                let mut width = mem::uninitialized();
+                let mut height = mem::uninitialized();
+                let _ = ffi::gdk_pixbuf_get_file_info_finish(res, &mut width, &mut height, &mut error);
+                let result = if error.is_null() { Ok((width, height)) } else { Err(from_glib_full(error)) };
+                let callback: &&(Fn(Result<(i32, i32), Error>) + Send + Sync + 'static) = transmute(user_data);
+                callback(result);
+            }
+        }
+        let callback = get_file_info_async_trampoline;
+        unsafe {
+            ffi::gdk_pixbuf_get_file_info_async(filename.to_glib_none().0, cancellable.0, Some(callback), Box::into_raw(user_data) as *mut _);
+        }
+    }
+
+    pub fn get_formats() -> Vec<PixbufFormat> {
+        unsafe {
+            FromGlibPtrContainer::from_glib_container(ffi::gdk_pixbuf_get_formats())
+        }
+    }
+
+    //pub fn new_from_stream_async<'a, P: IsA</*Ignored*/gio::InputStream>, Q: Into<Option<&'a gio::Cancellable>>, R: Fn(Result<(), Error>) + Send + Sync + 'static>(stream: &P, cancellable: Q, callback: R) {
```

Comment:
gio::InputStream, OutputStream and IOStream should be generatable at this point

Reply:
Your right, at minimum `gio::InputStream`, seems fully generated.
Same, IMHO this not part of this PR.

Reply:
I preparing PR for Gio